src/main/java/org/myrobotlab/service/Tracking.java

```diff
@@ -1,592 +1,395 @@
-  private String state = STATE_IDLE;
-
-  // ------ PEER SERVICES BEGIN------
-  transient public Pid pid;
-  transient public OpenCV opencv;
-
-  // transient public ServoControl x, y;
-  private class TrackingServoEvent {
-    transient ServoControl servoControl = null;
-    int scanStep = 2;
-    Double currentServoPos;
-    String name;
-    String axis;
-
-    TrackingServoEvent(String name) {
-      this.name = name;
-    }
-  }
+  public final static Logger log = LoggerFactory.getLogger(Tracking.class);
+
+  // FIXME - enable / disable
+  List<String> trackingFilters = new ArrayList<>();
 
-  transient private HashMap<String, TrackingServoEvent> servoControls = new HashMap<String, TrackingServoEvent>();
+  // FIXME - should just be publishing a normalized x,y polar coordinate
+  // and let the control system deal with that - having "servos" is
+  // too device specific
 
-  // statistics
-  public int updateModulus = 1;
+  PidData xPid = new PidData();
 
-  public long cnt = 0;
+  PidData yPid = new PidData();
 
-  public Point2df lastPoint = new Point2df(0.5F, 0.5F);
+  String pan;
 
-  double sizeIndexForBackgroundForegroundFlip = 0.10;
+  String tilt;
 
-  int faceFoundFrameCount = 0;
+  String pid;
 
-  int faceFoundFrameCountMin = 2;
+  String cv;
 
-  boolean scan = false;
+  long lostTrackingDelayMs = 1000;
+
+  enum TrackingState {
+    IDLE, SEARCHING, TRACKING
+  }
+
+  public static class Stats {
+    public int inputCnt = 0;
+    public float latency = 0;
+  }
 
-  String[] axis = new String[] { "x", "y" };
+  TrackingState state = TrackingState.IDLE;
 
-  // FIXME !! question remains does the act of creating meta update the
-  // reservatinos ?
-  // e.g if I come to the party does the reservations get updated or do I
-  // crash the party ??
-  public Tracking(String n, String id) throws Exception {
+  Stats stats = new Stats();
+
+  public Tracking(String n, String id) {
     super(n, id);
 
-    pid = (Pid) createPeer("pid");
-    // the kp should be proportional to the input min/max of the servo.. for now
-    // we'll go with 45 for now.
-    pid.setPID("x", 3.0, 1.0, 0.1);
-    pid.setControllerDirection("x", Pid.DIRECTION_DIRECT);
-    pid.setMode("x", Pid.MODE_AUTOMATIC);
-    pid.setOutputRange("x", -5, 5); // <- not correct - based on maximum
-    pid.setSampleTime("x", 30);
-    pid.setSetpoint("x", 0.5); // set center
-
-    pid.setPID("y", 3.0, 1.0, 0.1);
-    pid.setControllerDirection("y", Pid.DIRECTION_DIRECT);
-    pid.setMode("y", Pid.MODE_AUTOMATIC);
-    pid.setOutputRange("y", 5, -5); // <- not correct - based on maximum
-    pid.setSampleTime("y", 30);
-    pid.setSetpoint("y", 0.5); // set center
+    registerForInterfaceChange(ServoControl.class);
+    registerForInterfaceChange(PidControl.class);
+    registerForInterfaceChange(ComputerVision.class);
 
-  }
+    // defaults - pre getConfig or load ?
+    trackingFilters.add("FaceDetectDNN");
+
+    // publishes state/status info from this service

```

Comment:
why is this in the constructor?  it should be in the start service lifecycle instead.